File: palindrome/9. Palindrome Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        temp = x
        y = 0 # y 是 x 翻转后的数字。比较 y == x 就可以知道x是不是回文数字了
        while temp > 0:
            last_num = temp % 10
            temp //= 10
            y = y * 10 + last_num
        
        return y == x

File: palindrome/9_palindrome_number.rs
struct Solution;

impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x < 0 {
            return false;
        }
        let mut temp: i32 = x;
        let mut reversed = 0;

        while temp > 0 {
            let rightmost_digit = temp % 10;
            temp = temp / 10;
            reversed = reversed * 10 + rightmost_digit;
        }

        x == reversed
    }
}

fn main() {
    assert!(Solution::is_palindrome(123) == false);
    assert!(Solution::is_palindrome(121) == true);
    assert!(Solution::is_palindrome(-11) == false);
}